internal/alloy/logging/handler.go

```diff
@@ -141,54 +92,126 @@ func (h *handler) buildHandler() slog.Handler {
 		panic(fmt.Sprintf("unknown format %v", expectFormat))
 	}
 
-	newHandler = newHandler.WithAttrs(h.attrs)
+	// Need to replay our groups and attrs in the correct order.
+	for _, n := range h.nested {
+		if n.group != "" {
+			newHandler = newHandler.WithGroup(n.group)
+		} else {
+			newHandler = newHandler.WithAttrs(n.attrs)
+		}
+	}
 
 	h.currentFormat = expectFormat
 	h.inner = newHandler
 	return newHandler
 }
 
 func (h *handler) WithAttrs(attrs []slog.Attr) slog.Handler {
-	newAttrs := slices.Clone(h.attrs)
-
-	if len(h.group) > 0 {
-		// Deeply nest the attributes under the appropriate group.
-		groupAttr := slog.Attr{
-			Key:   h.group[len(h.group)-1],
-			Value: slog.GroupValue(attrs...),
-		}
-		rem := h.group[:len(h.group)-1]
-
-		for len(rem) > 0 {
-			groupAttr = slog.Attr{
-				Key:   rem[len(rem)-1],
-				Value: slog.GroupValue(groupAttr),
-			}
-			rem = rem[:len(rem)-1]
-		}
-
-		newAttrs = append(newAttrs, groupAttr)
-	} else {
-		newAttrs = append(newAttrs, attrs...)
-	}
+	newNest := make([]nesting, len(h.nested)+1)
+	newNest = append(newNest, h.nested...)
+	newNest = append(newNest, nesting{
+		attrs: attrs,
+	})
```

Comment:
This isn't correct; it's allocating `len(h.nested)+1` elements but then appending `len(h.nested)+1` _new_ elements to the end of the array, so there's a total of `2*(len(h.nested)+1)` elements at the end:

https://go.dev/play/p/ajPdFQmWOG_p